docs/DeveloperGuide.md

```diff
@@ -310,45 +310,333 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
+
+2. LibTask shows the list of all the patrons.
+
+   Use case ends.
+
+**Extensions**
+
+* 2a. LibTask has no stored patrons.
+
+    * 2a1. LibTask shows an empty list.
+
+      Use case ends.
+
+### UC03: Editing a patron on LibTask
+
+**MSS**
+1. User list all patrons [UC02](#uc02-list-patrons-on-libtask)
+
+2. User requests to edit a patron and provide the index and the necessary details to be edited.
+
+3. LibTask edits the patron.
+
+   Use case ends.
+
+**Extensions**
+
+* 1a. The index is not provided or is invalid.
+
+    * 1a1. LibTask shows an error message.
+
+      Use case resumes from step 1.
+
+* 1b. The details are not provided or are invalid.
+
+    * 1b1.  LibTask shows an error message.
+
+      Use case resume from step 1.
+
+
+###UC04: Find a patron on LibTask
+
+**MSS**
+
+1. User request to find patron(s) and provide a number of keywords.
+
+2. LibTask shows the list of patrons that match the search.
+
+   Use case ends.
+
+**Extensions**
+
+* 1a. Keywords not provided or are invalid.
+
+    * 1a1. LibTask shows an error message.
+
+      Use case resumes from step 1.
+
+* 2a.  No patrons fulfil the search criteria.
+
+    * 2a1. LibTask shows an empty list.
+
+      Use case ends.
+
+
+**UC05: Delete a patron from LibTask**
+
+**MSS**
+
+1. User list all patrons [UC03](#uc02-list-patrons-on-libtask).
+
+2. User requests to delete a patron and provide the index.
+
+3. LibTask deletes the patron.
+
+   Use case ends.
+
+**Extensions**
+
+* 1a. The index is not provided or is invalid.
+
+    * 1a1. LibTask shows an error message.
+
+      Use case resumes from step 1.
+
+###UC06: Add book to LibTask
+
+**MSS**
+
+1. User requests to add a book and provides the name and details of the book
+
+2. LibTask adds the module.
+
+   Use case ends.
+
+**Extensions**
+
+* 1a. Compulsory fields are not provided.
+
+    * 1a1. LibTask shows an error message.
+
+      Use case resumes from step 1.
+
+* 1b. The book details are invalid.
+
+    * 1b1. LibTask shows an error message.
+
+      Use case resumes from step 1.
+
+###UC07: List Books on LibTask
+
+**MSS**
+1. User requests to list all books.
+
+2. LibTask shows the list of all the books.
+
+   Use case ends.
+
+###UC08: Edit a book on LibTask
+
+**MSS**
+
+1. User requests to edit a book and provides the index of the book and the new details.
+
+2. LibTask edits the book.
+
+   Use case ends.
+
+**Extensions**
+
+* 1a. The given index is invalid.
+
+    * 1a1. LibTask shows an error message.
+
+      Use case resumes from step 1.
+
+* 1b. The new details are invalid.
+
+    * 1b1. LibTask shows an error message.
+
+      Use case resumes from step 1.
+
+###UC09: Delete Book from LibTask
+
+**MSS**
+
+1. User requests to delete a book and provides the index.
+
+2. LibTask deletes the book.
+
+   Use case ends.
+
+**Extensions**
+
+* 1a. The given index is invalid.
+
+    * 1a1. LibTask shows an error message.
+
+      Use case resumes from step 1.
+
+###UC10: Borrow Book
```

Comment:
Similar to the issue raised [above](https://github.com/AY2122S2-CS2103T-W14-1/tp/pull/30/files#r817409999).